javascript/javascript1/week3/script.js

```diff
@@ -0,0 +1,147 @@
+
+//Exercise 1
+
+const names = [
+    "Peter",
+    "Ahmad",
+    "Yana",
+    "kristina",
+    "Rasmus",
+    "Samuel",
+    "katrine",
+    "Tala",
+];
+
+const nameToRemove = "Ahmad";
+
+function removeName(names, nameToRemove) {
+    const remove = names.indexOf(nameToRemove);
+    if (remove !== -1) {
+        names.splice(remove, 1);
+    }
+}
+
+removeName(names, nameToRemove);
+
+console.log(names); // ['Peter', 'Yana', 'kristina', 'Rasmus', 'Samuel', 'katrine', 'Tala']
+
+
+//2nd Exercise
+
+const travelInformation = {
+    speed: 50,
+    destinationDistance: 432,
+};
+function travelCalculation(travelInformation) {
+    const timeInHours = (travelInformation.destinationDistance / travelInformation.speed);
+    const hours = Math.floor(timeInHours);
+    const minutes = Math.round((timeInHours - hours) * 60);
+
+    return hours + " hours " + "and " + minutes + " minutes";
+
+
+}
+
+const travelTime = travelCalculation(travelInformation);
+console.log(travelTime); // 8 hours and 38 minutes
+
+
+//3rd Exercise 
+
+const seriesDurations = [
+    {
+        title: "Game of thrones",
+        days: 3,
+        hours: 1,
+        minutes: 0,
+    },
+    {
+        title: "Sopranos",
+        days: 3,
+        hours: 14,
+        minutes: 0,
+    },
+    {
+        title: "The Wire",
+        days: 2,
+        hours: 12,
+        minutes: 0,
+    },
+];
+
+
+function logOutSeriesText() {
+    const averageLifespan = ((80 * 365 * 24 * 60) / 100); // assuming 80 years
+    let totalPercentage = 0;
+    seriesDurations.forEach((series) => {
```

Comment:
Hi, the mix is intentional because I'm trying to implement and solve the exercises in different ways to challenge myself but also to understand truly the reason for the code if that makes sense.

From what I understand, forEach is similar to a for loop but is better suited when I want to process or calculate something for each element in the same array. The => syntax represents a function that processes each element in the array.

For example, in seriesDurations.forEach((series) => { ... }), the code processes each series object in the seriesDurations array, calculating the time, percentage, or any other operation for every series.

Of course, this is how I understood it so far please correct me if I'm mistaken.

Reply:
Thanks for the explanation! I would say you are pretty spot on. Although `forEach` is not super suited for processing data. In these cases you probably want to use map, filter, reduce, etc. It is mostly for "side effects" such as logging or sending data, i.e. affecting things outside the return value. We are getting a bit ahead of ourselves here so it is fine if this does not make sense. 😊